src/main/webapp/app/file-upload-assessment/file-upload-assessment.component.ts

```diff
@@ -0,0 +1,482 @@
+        this.fileUploadSubmissionService.getFileUploadSubmissionForExerciseWithoutAssessment(exerciseId, true).subscribe(
+            (submission: FileUploadSubmission) => {
+                this.handleReceivedSubmission(submission);
+                // Update the url with the new id, without reloading the page, to make the history consistent
+                const newUrl = window.location.hash.replace('#', '').replace('new', `${this.submission!.id}`);
+                this.location.go(newUrl);
+            },
+            (error: HttpErrorResponse) => {
+                if (error.status === 404) {
+                    // there is no submission waiting for assessment at the moment
+                    this.goToExerciseDashboard();
+                    this.jhiAlertService.info('artemisApp.tutorExerciseDashboard.noSubmissions');
+                } else if (error.error && error.error.errorKey === 'lockedSubmissionsLimitReached') {
+                    this.goToExerciseDashboard();
+                } else {
+                    this.onError('artemisApp.assessment.messages.loadSubmissionFailed');
+                }
+            },
+        );
+    }
+
+    private loadSubmission(submissionId: number): void {
+        this.fileUploadSubmissionService
+            .get(submissionId)
+            .pipe(filter(res => !!res))
+            .subscribe(
+                res => {
+                    this.handleReceivedSubmission(res.body!);
+                    this.loadFeedbacks(this.result.feedbacks);
+                },
+                (error: HttpErrorResponse) => {
+                    if (error.error && error.error.errorKey === 'lockedSubmissionsLimitReached') {
+                        this.goToExerciseDashboard();
+                    } else {
+                        this.onError('');
+                    }
+                },
+            );
+    }
+
+    private handleReceivedSubmission(submission: FileUploadSubmission): void {
+        this.submission = submission;
+        this.participation = this.submission.participation as StudentParticipation;
+        this.exercise = this.participation.exercise as FileUploadExercise;
+        this.result = this.submission.result;
+        if (this.result.hasComplaint) {
+            this.getComplaint();
+        }
+        this.submission.participation.results = [this.result];
+        this.result.participation = this.submission.participation;
+        if (this.result.feedbacks) {
+            this.loadFeedbacks(this.result.feedbacks);
+        } else {
+            this.result.feedbacks = [];
+        }
+        if ((this.result.assessor == null || this.result.assessor.id === this.userId) && !this.result.completionDate) {
+            this.jhiAlertService.clear();
+            this.jhiAlertService.info('artemisApp.assessment.messages.lock');
+        }
+
+        this.formattedGradingInstructions = this.artemisMarkdown.htmlForMarkdown(this.exercise.gradingInstructions);
```

Comment:
You could get rid of the attributes if you would use the `renderMarkdown` pipe in the template.

Reply:
True, looks cleaner now